### PR TITLE
native: harden subscriptions

### PR DESCRIPTION
**Pull request by latter-bolden**
- recovers from subscription `quit` events initiated by the client
- handles fatal client errors by rerunning our sync start logic 
- adds additional logging for client events

This is likely what was biting us if you lost connection randomly, in particular if you left the app open overnight and checked it the next day. If Urbit automatically closes the connection after ~12 hours of inactivity, we'll properly detect that and resub.

Tested by connecting to a local ship I could start/stop as needed. Tested with full & seamless resets.

Fixes LAND-1857

### CODE REVIEW

**Comment by latter-bolden** [comment]
After talking with @dnbrwstr, the strategy we want to try for keeping the cache synchronized is:

- rely on the subscription for real time updates while you're using the app or it's in the background
- for ensuring no changes get dropped when the app is closed or unexpected network behavior occurs (client hard reset), we'll leverage the new `/changes` scry and trigger it defensively where needed (upon navigating to a channel for example)

This hopefully gives us a good balance without requiring complicated catchup logic. Once that new scry makes it in we'll try connecting things up and see how it feels.

In the meantime, I think this can be merged. Only change in behavior is ensuring the subscriptions stay alive over time.

File: apps/tlon-mobile/src/components/AuthenticatedApp.tsx [hunk 1]
```diff
@@ -19,11 +19,14 @@ function AuthenticatedApp({ initialNotificationPath }: AuthenticatedAppProps) {
   useDeepLinkListener();
 
   useEffect(() => {
-    configureClient(ship ?? '', shipUrl ?? '');
-    sync.start().catch((e) => {
-      console.warn('Sync failed', e);
-    });
-    useStorage.getState().start();
+    const start = () => {
```
Comment (dnbrwstr) [inline apps/tlon-mobile/src/components/AuthenticatedApp.tsx:22]: This wrapper function seems unnecessary? Or if it is necessary, I don't understand why, and a comment might help ha.

File: packages/shared/src/api/urbit.ts [hunk 2]
```diff
@@ -29,11 +29,13 @@ export function configureClient({
   shipUrl,
   fetchFn,
   verbose,
+  onReset,
```
Comment (dnbrwstr) [inline packages/shared/src/api/urbit.ts:32]: Doesn't seem like this is actually used?

File: packages/shared/src/api/urbit.ts [hunk 3]
```diff
@@ -43,8 +45,21 @@ export function configureClient({
     logger.log('status-update', status);
   });
 
+  clientInstance.onReconnect = () => {
+    logger.log('client reconnect');
+  };
+
+  clientInstance.on('reset', () => {
+    logger.log('client reset');
```
Comment (dnbrwstr) [inline packages/shared/src/api/urbit.ts:53]: Will we ever have to do anything in response to these reset events?

**Review by dnbrwstr — COMMENTED** [review]
Looking good, a few questions.

**Review by dnbrwstr — APPROVED** [review]
Makes more sense after discussion, questions are nonblocking.

**Pull request merged**